core/src/types/operator/builder.rs

```diff
@@ -371,6 +371,11 @@ impl<A: Accessor> OperatorBuilder<A> {
     pub fn finish(self) -> Operator {
         let ob = self.layer(TypeEraseLayer);
 
-        Operator::from_inner(Arc::new(ob.accessor) as FusedAccessor)
+        let op = Operator::from_inner(Arc::new(ob.accessor) as FusedAccessor);
+        if !op.info().can_blocking() {
+            op.layer(BlockingLayer::default())
```

Comment:
blocking layer should not be enabled by default since it will introduce  cost for users don't want this. We can enable blocking layer in behavior tests directly for test.